src/libraries/xapplication/src/xapplication.cpp

```diff
@@ -313,7 +319,35 @@ int32_t xtop_application::handle_register_node(std::string const & node_addr, st
     return store::xstore_success;
 }
 
-void xtop_application::create_thread_pools() {}
+bool xtop_application::update_node_size(uint64_t & node_size) {
```

Comment:
for error handling, use std::error_code. use unified way to report error. the corresponding callback signature should be changed too.

Reply:
done